druid-shell/src/platform/gtk/window.rs

```diff
@@ -695,14 +696,91 @@ fn get_modifiers(modifiers: gdk::ModifierType) -> keyboard::KeyModifiers {
 }
 
 fn make_key_event(key: &EventKey, repeat: bool) -> keyboard::KeyEvent {
+    use gdk::enums::key as gdk_key;
     let keyval = key.get_keyval();
-    let hardware_keycode = key.get_hardware_keycode();
 
-    let keycode = hardware_keycode_to_keyval(hardware_keycode).unwrap_or(keyval);
-
-    let text = gdk::keyval_to_unicode(keyval);
-
-    keyboard::KeyEvent::new(keycode, repeat, get_modifiers(key.get_state()), text, text)
+    // KeyCodes::NumPad* is based around the mac's hardware key representation
+    // which doesn't have a numlock
+    //
+    // In these cases, we want to map to a key which is *not* the lowest level key
+    // On PC Keyboards the lower level keys represent NumPadHome and friends.
+    // which don't exist in `keycodes::KeyCode` so we special case these here.
+    //
+    // We can't just do this in hardware_keycode_to_keyval by forcing an interpretation
+    // with Numlock on there since this is unlikely to work laptop keyboards.
+    //
+    // So we do 2 things:
+    // * If the keyval is a NumPad[0-9], decimal or separator, avoid getting the hardware keyval.
+    // * If the keyval is a NumPad{Home,...} We translate it to a NumPad[0-9].
```

Comment:
I find this pretty terrible, since it is essentially hard coding the correlation between KP_0, and KP_Home...
But haven't managed to successfully do anything else with the gtk backend.